File: Backend/ITSupportAPI/Models/TicketHistory.cs
using System;
using System.Collections.Generic;

namespace ITSupportAPI.Models;

public partial class TicketHistory
{
    public int Id { get; set; }

    public int TicketId { get; set; }

    public int UserId { get; set; }

    public int? OldStatusId { get; set; }

    public int? NewStatusId { get; set; }

    public string Action { get; set; } = null!;

    public string? Notes { get; set; }

    public DateTime CreatedAt { get; set; }

    public virtual Status? NewStatus { get; set; }

    public virtual Status? OldStatus { get; set; }

    public virtual Ticket Ticket { get; set; } = null!;

    public virtual User User { get; set; } = null!;
}

File: Frontend/it-support-frontend/src/app/services/ticket-history.ts
import { Injectable } from '@angular/core';

import {
  HttpClient,
  HttpHeaders
} from '@angular/common/http';

import { Observable } from 'rxjs';


export interface TicketHistoryItem {

  id: number;

  ticketId: number;

  userId: number;

  userName: string;

  oldStatusId: number | null;

  oldStatusName: string | null;

  newStatusId: number | null;

  newStatusName: string | null;

  action: string;

  notes: string | null;

  createdAt: string;

}


@Injectable({
  providedIn: 'root'
})
export class TicketHistory {

  private apiUrl =
    'http://localhost:5058/api/tickethistory';


  constructor(
    private http: HttpClient
  ) {}


  getTicketHistory():
    Observable<TicketHistoryItem[]> {

    const token =
      localStorage.getItem('token');


    const headers = new HttpHeaders({
      Authorization: `Bearer ${token}`
    });


    return this.http.get<TicketHistoryItem[]>(
      this.apiUrl,
      { headers }
    );

  }

}